problem1.rb

```diff
@@ -0,0 +1,149 @@
+class Station
+
+  attr_reader :name, :trains
+
+  def initialize(name)
+    @name = name
+    @trains = []
+  end
+
+  def get_train(train)
+    self.trains << train
+  end
+
+  def send_train(train)
+    self.trains.delete(train)
+  end
+
+  def trains_type(type)
+    list = Array.new
+    self.trains.each do |train|
+      if train.type == type
+        list.append(train)
+      end
+    end
+    return list
+  end
+
+end
+
+class Route
+
+  attr_reader :stations
+
+  def initialize(first_station, last_station)
+    @stations = [first_station, last_station]
+  end
+
+  def add_station(station)
+    @stations.insert(-2, station)
+  end
+
+  def del_station(station)
+    @stations.delete(station)
+  end
+
+  def first_station
+    @stations[0].name
```

Comment:
```suggestion
    @stations[0]
```
метод называется `first_station`, а не `first_station_name`, т.е. читающий код человек будет ожидать возврат объекта, а не аттрибута